storage/rocksdb/rdb_index_merge.cc

```diff
@@ -173,6 +173,18 @@ int Rdb_index_merge::add(const rocksdb::Slice &key, const rocksdb::Slice &val) {
       LogPluginErrMsg(ERROR_LEVEL, 0, "Error writing sort buffer to disk.");
       return HA_ERR_ROCKSDB_MERGE_FILE_ERR;
     }
+
+    /*
+      The unsorted buffer may be too small for the key-value pair.
+    */
+    const uint data_size = RDB_MERGE_CHUNK_LEN + RDB_MERGE_KEY_DELIMITER +
+                           RDB_MERGE_VAL_DELIMITER + key.size() + val.size();
+    if (data_size > m_rec_buf_unsorted->m_total_size) {
+      LogPluginErrMsg(ERROR_LEVEL, 0,
```

Comment:
Wouldn't it be more informative to print something like:
`Current value of rocksdb_merge_buf_size=ABC is too small. At least XYZ bytes required`

Reply:
Yes, you are right, it is better as users will get a clear action plan alongside with the error message.

I was in doubt whenever I should make changes to the error message as the message in the patch is a duplicate of exactly the same message half a dozen lines above. The error of overrunning a buffer is conceptually the same, but is checked twice. I'd expect both cases to produce the same error, so they are identical. It's possible to easily remove duplication, but the only way I found involved `goto` backwards, which is not immediately understandable. For example a reader (human) may ask if infinite looping is possible. It's not, but it would take time to prove that.

I've updated the message in both places in the code and updated the test.